pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -229,6 +289,16 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
 	return builder
 }
 
+func (configgen *ConfigGeneratorImpl) getListenerName(bind string, port int, transport istionetworking.TransportProtocol) string {
+	switch transport {
+	case istionetworking.TransportProtocolTCP:
+		return bind + "_" + strconv.Itoa(port)
+	case istionetworking.TransportProtocolQUIC:
+		return "quic_" + bind + "_" + strconv.Itoa(port)
```

Comment:
we may consider here, and many other places replacing quic with udp. The rationale is that in the future we may add non-quic UDP and it would be on the same listener?

Unless quic + non-quic UDP on the same listener will never be supported by envoy

Reply:
If I understand it correctly [RFC-9000](https://datatracker.ietf.org/doc/html/rfc9000#section-17.3.1) gives a hint on QUIC co-existing with other UDP based protocols on the same port. 

>    Fixed Bit:  The next bit (0x40) of byte 0 is set to 1.  Packets
      containing a zero value for this bit are not valid packets in this
      version and MUST be discarded.  A value of 1 for this bit allows
      QUIC to coexist with other protocols; see [RFC7983].

So I agree with you that although QUIC is connection-oriented protocol, it is still UDP now. So I changed it to `udp_` as you suggested. I know that these names show up on `istioctl proxy-config listeners`. Question - Is this something that needs to be backwards-compatible?

Reply:
> Question - Is this something that needs to be backwards-compatible?

No, but if you change it it would probably make upgrades not work because Envoy would complain it has two listeners on the same port. But aside from that users are not supposed to depend on XDS details

Reply:
Ok. I thought we send the entire state of the world on each LDS push - meaning, if the new update received does not have the name then envoy would clean up the listener. Hmm...That does not seem to be the case here? (I don't know the details of Envoy XDS implementation). Anyways, I changed it to `udp_` as suggested.

Reply:
We do but envoy doesn't actually atomically apply them like that IIUC. So the new ones still can conflict with old ones

Reply:
Oh! Ok. Learned something new :) I didn't know "does not atomically apply them" part. I guess even if it applies atomically, Envoy has to put old listeners to draining mode for deletion (socket still not closed at that time). For modification, simple way is delete+add - in this case we will run into port-conflict.

But still - while upgrading don't we restart pods for newer version of sidecar to be injected and it connects to newer revision of istiod. Due to restart, envoy should have forgotten configs and starts with clean slate. Right? I see that the problem occurs when someone upgrades it in-place (which I think is not recommended) and sidecar reconnects. Another possibility is somebody changing Istiod tag (due to say, hot fix) resulting in newer Istiod sending something (again - hot-fix should be avoided).

Anyways, I have changed it to `udp_` as you suggested already. Still not fully convinced with "upgrades does not work" part. I have outlined scenarios that I could think of and yes - I agree that it would be a problem in case of in-case upgrades as for the sidecar it would be "same istiod address and reconnect" resulting in mixing up of envoy config state. But for canary upgrade scenario it should not cause issues as the pod would be restarted to point to the new revision.